src/lib/journey/stages/generic.svelte

```diff
@@ -112,12 +112,13 @@
       {callback}
       callbackMetadata={callbackMetadataArray[idx]}
       selfSubmitFunction={determineSubmission}
+      {step}
       stepMetadata={{ ...stepMetadata }}
       style={$style}
     />
   {/each}
 
-  {#if !stepMetadata.isStepSelfSubmittable}
+  {#if !stepMetadata.isStepSelfSubmittable || stepMetadata.isStepProvidingLocalAuth}
```

Comment:
This is what I meant in my earlier comment about "_we still need a "Next" button if the Select IDP Node has the Local Authentication option enabled._". The Select IDP callbacks are self submittable, but otherwise don't play by the rules!

Reply:
As seen in the earlier comment, I'd like to keep use-case specific code out of the generic, high-level components like this one. Hopefully, my suggestion will help here.

Reply:
Just a thought as I'm reworking the changes based on your other comments:

Each stage has its own version of the login form and therefore "next" button, so any logic like the above will need to be duplicated between them. To keep things consistent in case for example we have a social flow as part of a registration journey, I'll need to make the same changes to the "registration" and "username-password" stage files. Similarly for the "redirect" functionality - each stage will have to implement this.

Is there a way that we can avoid duplication of code like this and therefore minimise risk of mistakes, in the eventuality that the number of stages grows?

Reply:
There's no need to worry about the other stages as they are just examples of a named stage of known callbacks. Only the `generic` stage needs to handle anything thrown at it.

Reply:
Ok makes sense. Thanks.